packages/graphql-codegen-cli/src/utils/strip-comments.ts

```diff
@@ -1,41 +1,46 @@
 import * as babylon from 'babylon';
+import { CommentBlock, CommentLine, Comment } from 'babel-types';
 
 /*!
  * strip-comments <https://github.com/jonschlinkert/strip-comments>
  *
  * Copyright (c) 2014-2016, 2018, Jon Schlinkert.
  * Released under the MIT License.
  *
- * Strip all code comments from the given `input`, including protected
- * comments that start with `!`, unless disabled by setting `options.keepProtected`
- * to true.
- *
- * ```js
- * const str = strip('const foo = "bar";// this is a comment\n /* me too *\/');
- * console.log(str);
- * // => 'const foo = "bar";'
- * ```
- * @name  strip
- * @param  {String} `input` string from which to strip comments
- * @param  {Object} `options` optional options, passed to [extract-comments][extract-comments]
- * @option {Boolean} [options] `line` if `false` strip only block comments, default `true`
- * @option {Boolean} [options] `block` if `false` strip only line comments, default `true`
- * @option {Boolean} [options] `keepProtected` Keep ignored comments (e.g. `/*!` and `//!`)
- * @option {Boolean} [options] `preserveNewlines` Preserve newlines after comments are stripped
- * @return {String} modified input
- * @api public
  */
 
-/**
- * Strip comments
- */
+const removeComment = (
+  str: string,
+  comment: CommentBlock | CommentLine | Comment,
+  pos: { start: number; end: number; removed: number }
+) => {
+  let nl = '';
+
+  // the casting is safe because if type does not exist
+  // we will be comparing undefined against a string
+  if ((comment as CommentLine).type === 'CommentLine') {
```

Comment:
You can use `is` function and avoid it:
```
function isCommentLine(obj: any): obj is CommentLine {
   return obj && obj.type && obj.type === 'CommentLine';
}
```

Reply:
Will do :)
I heavily appreciate this review, thanks.